[OSDOCS-20299]: CQA: HCP on bare metal disconnected (1 of 2) (#113856)
openshift/openshift-docs: modules/hcp-agentserviceconfig.adoc

[id="hcp-agentserviceconfig_{context}"]
= Deploying AgentServiceConfig resources

[role="_abstract"]

> The `AgentServiceConfig` custom resource is an essential component of the Assisted Service add-on that is part of {mce-short}. It is responsible for bare-metal cluster deployment. When the add-on is enabled, you deploy the `AgentServiceConfig` resource to configure the add-on.

Reviewer: ```suggestion
The `AgentServiceConfig` custom resource (CR) is an essential component of the Assisted Service add-on that is part of {mce-short}. The CR is responsible for bare-metal cluster deployment. When the add-on is enabled, you deploy the `AgentServiceConfig` CR to configure the add-on.
```
"It" threw me for a sec since we had a second noun there right before it. (ha)

Author: Nice suggestion!

In addition to configuring the `AgentServiceConfig` resource, you need to include additional config maps to ensure that {mce-short} functions properly in a disconnected environment.

.Procedure

> . Configure the custom registries by adding the following config map, which includes the disconnected details to customize the deployment:

Reviewer: ```suggestion
. Configure the custom registries by adding the following config map, which has the disconnected details to customize the deployment:
```
simplicity/minimalism/entirely subjective

Author: Vale told me to use "includes" instead of "contains", lol

    # ...
----
+

> Replace `dns.base.domain.name` with the DNS base domain name.

Reviewer: ```suggestion
Replace the `dns.base.domain.name` in the `registry.mirror` container reference with the DNS base domain name.
```
otherwise, this is all really in the `data.registries.conf` entry per the actual ConfigMap (I think it's easier to find the reference)

+
The object has two fields:

> * Custom CAs: This field has the Certificate Authorities (CAs) that are loaded into the various processes of the deployment.

Reviewer: ```suggestion
* Custom CAs: The `ca-bundle.crt` field has the Certificate Authorities (CAs) that are loaded into the various processes of the deployment.
```

> * Registries: The `Registries.conf` field has information about images and namespaces that need to be consumed from a mirror registry rather than the original source registry.

Reviewer: IMHO these two bullets should go first, in order of vertical appearance in the YAML (or better yet as concept elsewhere), and the base domain name should go after these

    resources:
      requests:
        storage: 20Gi
  osImages:
  - cpuArchitecture: x86_64
    openshiftVersion: "4.14"
    rootFSUrl: http://registry.dns.base.domain.name:8080/images/rhcos-414.92.202308281054-0-live-rootfs.x86_64.img
    url: http://registry.dns.base.domain.name:8080/images/rhcos-414.92.202308281054-0-live.x86_64.iso
    version: 414.92.202308281054-0
  - cpuArchitecture: x86_64
    openshiftVersion: "4.15"
    rootFSUrl: http://registry.dns.base.domain.name:8080/images/rhcos-415.92.202403270524-0-live-rootfs.x86_64.img
    url: http://registry.dns.base.domain.name:8080/images/rhcos-415.92.202403270524-0-live.x86_64.iso
    version: 415.92.202403270524-0
----
+

> * `metadata.annotations["unsupported.agent-install.openshift.io/assisted-service-configmap"]` references the config map name that the Operator consumes to customize behavior.

Reviewer: ```suggestion
* The `metadata.annotations["unsupported.agent-install.openshift.io/assisted-service-configmap"]` annotation references the config map name that the Operator consumes to customize behavior.
```
I prefer using the article/param/noun approach. Not sure why it was removed?

Author: I think I was following the example in the SSG a little too closely: https://redhat-documentation.github.io/supplementary-style-guide/#explain-commands-variables-in-code-blocks

I see now that the guidelines say that the examples don't have to be followed exactly.

> * `spec.mirrorRegistryRef.name` points to the config map that has disconnected registry information that the Assisted Service Operator consumes. This config map adds those resources during the deployment process.

Reviewer: ```suggestion
* `spec.mirrorRegistryRef.name` points to the `ConfigMap` CR that has the disconnected registry information that the Assisted Service Operator consumes. This config map adds those resources during the deployment process.
```
because we are clearly talking about a specific config map object here
